### PR TITLE
hostnamectl: show product uuid/hardware serial in regular output, plus various fixes

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/shared/bus-util.c [hunk 1]
```diff
@@ -754,3 +754,23 @@ int bus_query_sender_pidref(
...
         return bus_creds_get_pidref(creds, ret);
 }
+
+int bus_message_read_id128(sd_bus_message *m, sd_id128_t *ret) {
+        const void *a;
+        size_t sz;
+        int r;
+
+        assert(m);
+
```
Comment (bluca) [inline src/shared/bus-util.c:764]: assert(ret)

**Review by bluca — APPROVED** [review]

File: src/hostname/hostnamed.c [hunk 2]
```diff
@@ -263,94 +263,121 @@ static int get_hardware_firmware_data(const char *sysattr, char **ret) {
...
 }
 
 static int get_hardware_serial(char **ret) {
-         int r;
+        _cleanup_free_ char *b = NULL;
+        int r = 0;
+
+        FOREACH_STRING(attr, "product_serial", "board_serial") {
+                r = get_hardware_firmware_data(attr, &b);
+                if (r > 0)
```
Comment (YHNdnzj) [inline src/hostname/hostnamed.c]: Maybe, `!IN_SET(r, 0, -ENOENT)` instead?

**Review by YHNdnzj — APPROVED** [review]

**Comment by bluca** [comment]
The CI was red here? And it is definitely related:

```
autopkgtest [23:01:52]: test hostnamed: [-----------------------18:06
original hostname: autopkgtest-lxc-zivoio18:06
Failed to query hardware serial, ignoring: No such file or directory18:06
Failed to query hardware serial, ignoring: No such file or directory18:06
Failed to query hardware serial, ignoring: No such file or directory18:06
autopkgtest [23:01:52]: test hostnamed: -----------------------]18:06
hostnamed            FAIL stderr: Failed to query hardware serial, ignoring: No such file or directory18:06
autopkgtest [23:01:52]: test hostnamed:  - - - - - - - - - - results - - - - - - - - - -18:06
autopkgtest [23:01:53]: test hostnamed:  - - - - - - - - - - stderr - - - - - - - - - -18:07
Failed to query hardware serial, ignoring: No such file or directory18:07
Failed to query hardware serial, ignoring: No such file or directory18:07
Failed to query hardware serial, ignoring: No such file or directory
```

**Comment by poettering** [comment]
Uh, it had passed earlier. I am super confused by the new issue though, makes no sense to me at all...

**Comment by poettering** [comment]
Hmm, is there any way how we can actually get the debug output from that CI? it's just a dump of text lines, and no journal file, nothing? that makes it very hard to review.

my educated guess right now is that this is a PK issue, i.e. we propagate an error from PK improperly. (maybe it's simply missing in the test environment, and our fallback paths in that case don't work correctly?) but without any logs from hostnamed it's impossible to debug.

These test cases, where are they implemented? they are not part of our git tree, are they?

**Comment by bluca** [comment]
I don't think it passed? History shows both runs were red: https://the-real-systemd.semaphoreci.com/branches/11b4f9cc-2cf8-4f7a-b800-f729ac13987d

The test case is here: https://salsa.debian.org/systemd-team/systemd/-/blob/debian/master/debian/tests/hostnamed?ref_type=heads
It's just a very simple smoke test, and polkit is not installed, no

**Pull request merged**